lib/reek/code_comment.rb

```diff
@@ -7,10 +7,32 @@ module Reek
   # module, class and method definitions.
   #
   class CodeComment
-    CONFIGURATION_REGEX            = /:reek:(\w+)(:\s*\{.*?\})?/
+    CONFIGURATION_REGEX = /
+                          # 1st group is the smell name e.g.:
```

Comment:
>> I'm having a hard time distinguishing between code and comments here. I think this will be clearer if you put the large explanation above the constant, and the rest in trailing comments, like so:

Sure, I can change this.

>> I also have no idea what the first ? is doing in the second group.

You mean the 

>> ( \{.*?\} )?  # optional details in hash style

part?

That's for greedy matching, so in case you get stuff like

```
{ holla: waldfee } omg: bar }
````

just using "(.*)" would match the whole expression when we actually want the closest matching bracket - that's what "(.*?)" does :)